pallets/pools-registry/src/lib.rs

```diff
@@ -0,0 +1,395 @@
+	) -> DispatchResult;
+
+	fn update(
+		pool_id: PoolId,
+		changes: PoolChanges<Rate, MaxTokenNameLength, MaxTokenSymbolLength, MaxTranches>,
+	) -> DispatchResultWithPostInfo;
+
+	fn execute_update(pool_id: PoolId) -> DispatchResultWithPostInfo;
+}
+
+type PoolMetadataOf<T> = PoolMetadata<<T as Config>::MaxSizeMetadata>;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		type Balance: Member
+			+ Parameter
+			+ AtLeast32BitUnsigned
+			+ Default
+			+ Copy
+			+ MaxEncodedLen
+			+ FixedPointOperand
+			+ From<u64>
+			+ From<u128>
+			+ TypeInfo
+			+ TryInto<u64>;
+
+		type PoolId: Member
+			+ Parameter
+			+ Default
+			+ Copy
+			+ HasCompact
+			+ MaxEncodedLen
+			+ core::fmt::Debug;
+
+		type Rate: Parameter + Member + MaybeSerializeDeserialize + FixedPointNumber + TypeInfo;
+
+		/// A fixed-point number which represents an
+		/// interest rate.
+		type InterestRate: Member
+			+ Parameter
+			+ Default
+			+ Copy
+			+ TypeInfo
+			+ FixedPointNumber<Inner = Self::Balance>;
+
+		type ModifyPool: PoolMutate<
+			Self::AccountId,
+			Self::Balance,
+			Self::PoolId,
+			Self::CurrencyId,
+			Self::Rate,
+			Self::MaxTokenNameLength,
+			Self::MaxTokenSymbolLength,
+			Self::MaxTranches,
+		>;
+
+		type CurrencyId: Parameter + Copy;
+
+		type TrancheId: Member
+			+ Parameter
+			+ Default
+			+ Copy
+			+ MaxEncodedLen
+			+ TypeInfo
+			+ From<[u8; 16]>;
+
+		/// The origin permitted to create pools
+		type PoolCreateOrigin: EnsureOrigin<Self::Origin>;
+
+		/// Max length for a tranche token name
+		#[pallet::constant]
+		type MaxTokenNameLength: Get<u32> + Copy + Member + scale_info::TypeInfo;
+
+		/// Max length for a tranche token symbol
+		#[pallet::constant]
+		type MaxTokenSymbolLength: Get<u32> + Copy + Member + scale_info::TypeInfo;
+
+		/// Max number of Tranches
+		#[pallet::constant]
+		type MaxTranches: Get<u32> + Member + scale_info::TypeInfo;
+
+		/// Max size of Metadata
+		#[pallet::constant]
+		type MaxSizeMetadata: Get<u32> + Copy + Member + scale_info::TypeInfo;
+
+		type Permission: Permissions<
+			Self::AccountId,
+			Scope = PermissionScope<Self::PoolId, Self::CurrencyId>,
+			Role = Role<Self::TrancheId, Moment>,
+			Error = DispatchError,
+		>;
+
+		/// Weight Information
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+	#[pallet::getter(fn get_pool_metadata)]
+	pub(super) type PoolMetadata<T: Config> =
+		StorageMap<_, Blake2_256, T::PoolId, PoolMetadataOf<T>>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A pool creation was registered
+		/// and forwarded to the pool-systems pallet
+		Registered { pool_id: T::PoolId },
```

Comment:
I think we want to already look-ahead with the events. Once a create is hitting `pools-registry`, it is not guaranteed yet that the creation will go through. We emit a `created` event inside `pool-systems`. 

I guess it's an in-between solution. I can see that we have to update Subgraph soon again because of all the new events we emit in both pallets. Therefore I am also open to change them to "as-is" and update them along the way.

Maybe @mustermeiszer can chime in as well?

Reply:
For events, we should try not to have in-between solutions. If we call this `Registered` now but change it to `Created` later once the code is fully ported, the subquery will forever have two event handlers. I think we should consider what the final event name would be, and just use that already now.